iModelCore/iModelPlatform/Tests/DgnProject/Compatibility/IModelCompatibilityTests.cpp

```diff
@@ -2071,3 +2071,200 @@ TEST_F(IModelCompatibilityTestFixture, UpgradeDomainIModelToEC32)
+            ASSERT_EQ(SchemaStatus::Success, dgnDb.ImportSchemas(deserializationCtx->GetCache().GetSchemas())) << testDbPtr->GetDescription();
+
+            // Perform major schema change by deleting property and class
+            auto deserializationCtxUpdate = TestFileCreator::DeserializeSchema(testDbPtr->GetDb(), SchemaItem(R"xml(<?xml version="1.0" encoding="utf-8" ?>
+                            <ECSchema schemaName="TestSchema" alias="ts" version="2.0" xmlns="http://www.bentley.com/schemas/Bentley.ECXML.3.1">
+                                <ECSchemaReference name="CoreCustomAttributes" version="01.00" alias="CoreCA" />
+                                <ECSchemaReference name="BisCore" version="01.00" alias="bis"/>
+                                
+                                <ECCustomAttributes>
+                                    <DynamicSchema xmlns = 'CoreCustomAttributes.01.00' />
+                                </ECCustomAttributes>
+
+                                <ECEnumeration typeName='UnstrictEnumInt' backingTypeName='int' isStrict='False'>
+                                    <ECEnumerator value = '0' displayLabel = 'txt' />
+                                    <ECEnumerator value = '1' displayLabel = 'bat' />
+                                </ECEnumeration>
+
+                                <ECEnumeration typeName='StrictEnumString' backingTypeName='string' isStrict='True'>
+                                    <ECEnumerator value = 'val10' displayLabel = 'txt' />
+                                    <ECEnumerator value = 'val11' displayLabel = 'bat' />
+                                </ECEnumeration>
+
+                                <ECEntityClass typeName="TestClass" >
+                                    <BaseClass>bis:PhysicalElement</BaseClass>
+                                    <ECProperty propertyName="PropStringToInt" typeName="int" />
+                                    <ECProperty propertyName="PropIntToString" typeName="string" />
+
+                                    <ECProperty propertyName="PropStringToEnum" typeName="UnstrictEnumInt" />
+                                    <ECProperty propertyName="PropIntToEnum" typeName="UnstrictEnumInt" />
+
+                                    <ECProperty propertyName="PropUnstrictEnumToString" typeName="string" />
+                                    <ECProperty propertyName="PropUnstrictEnumToInt" typeName="int" />
+                                    <ECProperty propertyName="PropStrictEnumToString" typeName="string" />
+                                    <ECProperty propertyName="PropStrictEnumToInt" typeName="int" />
+
+                                    <ECProperty propertyName="PropStrictEnumToUnstrictEnum" typeName="UnstrictEnumInt" />
+                                </ECEntityClass>
+                            </ECSchema>)xml"));
+            ASSERT_NE(deserializationCtxUpdate, nullptr) << testDbPtr->GetDescription();

```

Comment:
add a test with data and check and verify the type of data.

Reply:
Done !